src/3rdparty/squirrel/squirrel/sqstate.cpp

```diff
@@ -289,7 +291,9 @@ SQInteger SQSharedState::CollectGarbage(SQVM *vm)
 		t = t->_next;
 	}
 	_gc_chain = tchain;
+#ifndef NDEBUG
 	SQInteger z = _table(_thread(_root_vm)->_roottable)->CountUsed();
+#endif
 	assert(z == x);
```

Comment:
For readability I think it'd be better to wrap the `assert` into the `#ifndef` block too, even if it's the same effect.